### PR TITLE
Allow custom logos of any height in the admin menu

### PR DESCRIPTION
**Pull request by Meteor0id**
This provides a fix for Issue #3798 

Turns .inner into a flex container. The scrollbar is not compromised this way.

There is also PR #4746 which fixes various issues with other components of the nav bar.

### CODE REVIEW

**Comment by Meteor0id** [comment]
wagtail well aligned custom logo

Added two more commits to align the dashboard text to the center next to any custom logo.
Thi text only appears on lower resolutions.


Not that the decision of  how big the custom logo should be on mobile devices remains up to the admin himself. We cannot decide that for him, since we do not know how well his logo will look on small scale.
In my example the logo is much larger than the wagtail logo would have been, but as said, that is up to the admin.

**Comment by Meteor0id** [comment]
I have tested on desktop in firefox 61.0.2 at various resolutions. I also tested in on the android version of Firefox (also 61.0.2).

**Comment by Meteor0id** [comment]
I have tested on desktop in firefox 61.0.2 at various resolutions. I also tested in on the android version of Firefox (also 61.0.2).

File: wagtail/admin/static_src/wagtailadmin/scss/components/_logo.scss [hunk 1]
```diff
@@ -23,13 +23,20 @@
 }
 
 .logo {
-    display: block;
+    display: flex;
```
Comment (thibaudcolas) [inline wagtail/admin/static_src/wagtailadmin/scss/components/_logo.scss]: In my testing this makes the "Dashboard" text stick out of the nav.

wip-highlight-over-link

File: wagtail/admin/templates/wagtailadmin/base.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
 {% block furniture %}
     <div class="nav-wrapper">
         <div class="inner">
-            <a href="{% url 'wagtailadmin_home' %}" class="logo" title="Wagtail v.{% wagtail_version %}">
+            <a href="{% url 'wagtailadmin_home' %}" class="logo" title="{% trans "Dashboard" %}">
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/base.html]: We definitely want to keep this pointing at the version number.

**Comment by thibaudcolas** [comment]
I have rebased this with just the commits that fix the positioning of the logo. https://github.com/thibaudcolas/wagtail/pull/2 has a copy of the full original PR for reference.

File: wagtail/admin/static_src/wagtailadmin/scss/components/_main-nav.scss [hunk 3]
```diff
@@ -290,11 +295,7 @@ body.explorer-open {
     }
 
     .nav-main {
-        position: absolute;
-        top: 175px; // WARNING - magic number - the height of the logo plus search box
-        bottom: 0;
         margin-bottom: 127px; // WARNING - magic number - the height of the .footer
-        width: 100%;
```
Comment (thibaudcolas) [inline wagtail/admin/static_src/wagtailadmin/scss/components/_main-nav.scss]: As far as I can see none of these styles do anything with the current Flex implementation.

**Comment by thibaudcolas** [comment]
Thank you @Meteor0id! This will be released in Wagtail v2.4.

**Pull request merged**